db/db_impl/db_impl_open.cc

```diff
@@ -92,11 +92,12 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src) {
   if (result.recycle_log_file_num &&
       (result.wal_recovery_mode == WALRecoveryMode::kPointInTimeRecovery ||
        result.wal_recovery_mode == WALRecoveryMode::kAbsoluteConsistency)) {
-    // kPointInTimeRecovery is inconsistent with recycle log file feature since
-    // we define the "end" of the log as the first corrupt record we encounter.
+    // kPointInTimeRecovery is indistinguishable from
+    // kTolerateCorruptedTailRecords in recycle mode since we define
+    // the "end" of the log as the first corrupt record we encounter.
     // kAbsoluteConsistency doesn't make sense because even a clean
     // shutdown leaves old junk at the end of the log file.
-    result.recycle_log_file_num = 0;
+    result.wal_recovery_mode = WALRecoveryMode::kTolerateCorruptedTailRecords;
```

Comment:
The existing code here makes no sense to me either, tbh, but I'd rather incrementally improve on what we had before.